src/aitsi/m3spin/pkb/impl/ModifiesImpl.java

```diff
@@ -32,20 +32,20 @@ public void setModifies(Procedure proc, Variable var) {
             varsModifiedByProc.get(proc).add(var);
         }
         else{
-            List<Variable> varList = new ArrayList<>();
+            HashSet<Variable> varList = new HashSet<>();
             varList.add(var);
             varsModifiedByProc.put(proc,varList);
         }
     }
 
     @Override
     public List<Variable> getModified(Statement stmt) {
-        return varsModifiedByStmt.get(stmt);
+        return (List<Variable>) varsModifiedByStmt.get(stmt);
```

Comment:
Po zmianie zwracanego typu ten komentarz traci ważność, ale rzutowanie w tym miejscu nie ma sensu. Prawidłowo po return powinno być `new ArrayList<>(varsModifiedByStmt.get(proc))`. To samo w innych miejscach z podobnym rzutowaniem.

Reply:
Fixed